pkg/registry/apis/secret/storage.go

```diff
@@ -88,49 +102,69 @@ func checkPathOrValue(s *secret.SecureValue, mustExist bool) error {
 	return nil
 }
 
-func (s *secretStorage) Create(ctx context.Context,
+// Create a new `securevalue`. Does some validation and allows empty `name` (generated).
+func (s *secureValueRESTStorage) Create(
+	ctx context.Context,
 	obj runtime.Object,
+
+	// TODO: How to define this function? perhaps would be useful to keep all validation here and not use `checkPathOrValue` for example..
 	createValidation rest.ValidateObjectFunc,
+
+	// TODO: How can we use these options? Looks useful. `dryRun` for dev as well.
 	options *metav1.CreateOptions,
 ) (runtime.Object, error) {
 	sv, ok := obj.(*secret.SecureValue)
 	if !ok {
 		return nil, fmt.Errorf("expected SecureValue for create")
 	}
 
+	// TODO: path for key managers, value for everything else. actual schema for key managers might change so we need to update this as well.
 	err := checkPathOrValue(sv, true)
 	if err != nil {
 		return nil, err
 	}
 
+	// A `securevalue` may be created without a `name`, which means it gets generated on-the-fly.
 	if sv.Name == "" {
-		r, err := util.GetRandomString(8)
+		// TODO: how can we make sure there are no conflicts with existing resources?
+		generatedName, err := util.GetRandomString(8)
```

Comment:
this would be with existing resources in a namespace? the keyspace seems sufficient
Maybe a retry if there's a name conflict? uuids are quite long to use, loading them all and checking for uniqueness could be an option - TODO indeed

Reply:
Yeah I am not sure if we'd need to do a Get here after generating the name, or if we leave it to Kubernetes to fail the resource creation in the API when it finds a conflict.